app/build.gradle

```diff
@@ -113,6 +114,7 @@ dependencies {
     testImplementation "org.junit.jupiter:junit-jupiter-api:5.10.0"
     testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:5.10.0"
     testImplementation 'com.facebook.soloader:soloader:0.10.5'
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-rx2:1.9.0"
```

Comment:
Am I misunderstanding, or is this re-introducing RxJava2?
RxJava2 is "end of life" according to https://github.com/commons-app/apps-android-commons/issues/4665

Reply:
So, the function that actually makes the call is in our APIs file which is in java, so I couldn't really use coroutines without `suspend` as it is a kotllin syntax but I'm observing the result in `coroutine` way that's why I needed this rxjava to coroutine conversion.

see the comment to see where conversion happens.

Also, now that I see after updating my branch the the file is actually converted to .kt. So, do you want me to update my functions to use coroutines there too? or we can do conversion later, maybe all the calls in that file?


P.S: looking forward to adoption of coroutines :)

Reply:
> do you want me to update my functions to use coroutines there too?

That would be wonderful yes. 🙂

Reply:
Done :+1: